src/graph/build.rs

```diff
@@ -88,6 +88,11 @@ pub(crate) fn get_dep_graph(metadata: Metadata, config: &Config) -> anyhow::Resu
                         continue;
                     }
 
+                    // Don't add dependencies of dependencies if we're at the depth limit
+                    if depth + 1 > config.depth.unwrap_or(u32::MAX) {
```

Comment:
This should work too, right? Seems a bit more intuitive to me.

```suggestion
                    if config.depth.is_some_and(|max_depth| depth >= max_depth) {
```

Reply:
Yes thats definitely a better way to do it.